### ISSUE TITLE
$('html_value').text() is not safe.

### ISSUE THREAD

**Issue by 123Pro123Pro** [jquery/jquery#4924]
Hi 

I found that function $('html_value').text() is not safe. This function didn't completely remove all html tabs in a text.
For example:

```

<script src="https://code.jquery.com/jquery-3.6.0.js"></script>

<body>
    <input id="inp" value="<<script>script>console.log(1)</</script>script>">
    
    <div id="inp_div">

    </div>
</body>

<script type="text/javascript">
	var input_value = $("#inp").val();

	var input_remove_html_tab = $(input_value).text();  // --> input_remove_html_tab  = "<script>console.log(1)</script>"

	$("#inp_div").html(input_remove_html_tab);
</script>


```
`$(input_value).text();` didn't completely remove all html tabs in `input_value` 

Is there any patch for this function?

**Comment by wenz** [comment]
I may be wrong, but `text()` is not a sanitizing function, but basically returns `textContent` or `nodeValue` (for input, the forms -- but you may want to use val() anyway). If you want to safely output data, use `text()` to write a value to the DOM.

**Comment by mgol** [comment]
**tl;dr:** if you need to pass `.text()` output to the `.html(content)` setter, you need to first escape it.

**Longer explanation:** this is a very common gotcha; since both raw text & HTML are strings, people consider them interchangeable whereas in fact they are not. If you get text content from an element, you cannot assign it as HTML content of another element unless you escape it beforehand.

The `.text()` getter corresponds to the `.text(content)` setter, not to the `.html(content)` one. The following:
```js
elem2.text(elem1.text());
```
should make `elem2` have the same text content as `elem1`. If we were to escape HTML in the `.text()` output so that it's safe to pass to the `.html(content)` setter, the above correspondence would break. For example, for `elem1` defined via the following HTML:
```html
<div>&lt;br&gt;</div>
```
that shows `<br>` to the user invoking:
```js
elem2.text(elem1.text());
```
would make `elem2` show `&lt;br&gt;` to the user - the escaped content; clearly not what's intended.

BTW, you can avoid the issue of both text & HTML being strings by using Trusted HTML ([read more in this `web.dev` article](https://web.dev/trusted-types/)) which will not allow you to assign raw strings as HTML to an element. That said, the feature is only supported by Chromium and other browser engines don't plan to implement this.

**Comment by goarblt** [comment]
hello
https://www.evil.com
